src/concurrency/genmc/mod.rs

```diff
@@ -128,13 +129,42 @@ pub struct GenmcCtx {
 /// GenMC Context creation and administrative / query actions
 impl GenmcCtx {
     /// Create a new `GenmcCtx` from a given config.
-    fn new(miri_config: &MiriConfig, global_state: Arc<GlobalState>) -> Self {
+    fn new(miri_config: &MiriConfig, global_state: Arc<GlobalState>, mode: GenmcMode) -> Self {
         let genmc_config = miri_config.genmc_config.as_ref().unwrap();
-        let handle =
-            RefCell::new(create_genmc_driver_handle(&genmc_config.params, genmc_config.log_level));
+        let handle = RefCell::new(create_genmc_driver_handle(
+            &genmc_config.params,
+            genmc_config.log_level,
+            /* do_estimation: */ mode == GenmcMode::Estimation,
+        ));
         Self { handle, exec_state: Default::default(), global_state }
     }
 
+    /// Given the time taken for the estimation mode run,
+    /// print an estimation for how many executions the entire verification will require and give a total time estimate.
+    pub fn print_estimation_result(&self, elapsed_time: Duration) {
+        let result = self.handle.borrow().get_estimation_results();
+        let elapsed_time_sec = elapsed_time.as_secs_f64();
+        #[allow(clippy::as_conversions)]
+        let estimated_time_sec = elapsed_time_sec * result.mean as f64
+            / (u64::saturating_add(result.explored_execs, result.blocked_execs)) as f64;
```

Comment:
Does this match what GenMC does?

If you cast the `u64` to `f64` first you can do the addition without worrying about saturation/overflow.